File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i,j=0,1
        while i < len(nums):
            if j >= len(nums):
                i=i+1
                j=i+1
            if nums[i]+nums[j]==target:
                result = [i,j]
                return result
            else:
                j=j+1

File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        final List<List<Integer>> result = new ArrayList<>();
        Arrays.sort(nums);
        combination(result, new ArrayList<>(), nums, target, 0, 0);
        return result;
    }

    private static void combination(List<List<Integer>> result, List<Integer> curr, int[] nums,
        int target, int current, int i) {
        if (target == current) {
            result.add(new ArrayList<>(curr));
            return;
        }

        for (int j = i; j < nums.length; j++) {
            if (nums[j] + current > target) {
                return;
            }
            curr.add(nums[j]);
            combination(result, curr, nums, target, current + nums[j], j);
            curr.remove(curr.size() - 1);
        }
    }
}
